Description: Add "Ubuntu" and "Private Build" as recognized Java vendors
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug: https://bugs.documentfoundation.org/show_bug.cgi?id=124503
Bug-Ubuntu: https://launchpad.net/bugs/1822839

--- a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
+++ b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
@@ -250,7 +250,7 @@ public class LocalOfficeWindow
         if (getNativeWindowSystemType() == SystemDependent.SYSTEM_XWINDOW )
         {
             String vendor = System.getProperty("java.vendor");
-            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation") || vendor.equals("Debian"))
+            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation") || vendor.equals("Debian") || vendor.equals("Ubuntu") || vendor.equals("Private Build"))
                 && Boolean.getBoolean("sun.awt.xembedserver"))
             {
                 xembed = new NamedValue(
--- a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
+++ b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
@@ -28,6 +28,12 @@
     <vendor name="Debian">
       <minVersion>1.6.0</minVersion>
     </vendor>
+    <vendor name="Ubuntu">
+      <minVersion>1.6.0</minVersion>
+    </vendor>
+    <vendor name="Private Build">
+      <minVersion>1.6.0</minVersion>
+    </vendor>
     <vendor name="IBM Corporation">
       <minVersion>1.5.0</minVersion>
     </vendor>
--- a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
@@ -43,6 +43,8 @@ BEGIN_VENDOR_MAP()
     VENDOR_MAP_ENTRY("Sun Microsystems Inc.", SunInfo)
     VENDOR_MAP_ENTRY("Oracle Corporation", SunInfo)
     VENDOR_MAP_ENTRY("Debian", SunInfo)
+    VENDOR_MAP_ENTRY("Ubuntu", SunInfo)
+    VENDOR_MAP_ENTRY("Private Build", SunInfo)
     VENDOR_MAP_ENTRY("AdoptOpenJdk", SunInfo)
     VENDOR_MAP_ENTRY("Amazon.com Inc.", SunInfo)
 #ifndef MACOSX
